core/metrics-core-api/src/main/java/org/hawkular/metrics/core/api/MetricsService.java

```diff
@@ -84,6 +84,15 @@
 
     Observable<Metric> findMetrics(String tenantId, MetricType type);
 
+    /**
+     *
+     * @param tenantId
+     * @param tags
+     * @param type Optional MetricType, if null is given all matching metrics are returned
```

Comment:
I'm not sure what you mean actually. If a null is given, all the userTypes are used (the filter doesn't stop anything) but if a certain type is given, that one is used (and only metrics with that type are allowed to pass). There's a bug in that sense that this allow setting non-user-defined type also if you can bypass the REST-interface..